node/src/accountant/mod.rs

```diff
@@ -2769,6 +3754,724 @@ pub mod tests {
+        let system = System::new("pending_transaction");
+        let persistent_config = PersistentConfigurationMock::default().gas_price_result(Ok(130));
+        let blockchain_bridge = BlockchainBridge::new(
+            Box::new(blockchain_interface),
+            Box::new(persistent_config),
+            false,
+            Some(consuming_wallet),
+        );
+        let wallet_account_1 = make_wallet("creditor1");
+        let account_1 = PayableAccount {
+            wallet: wallet_account_1.clone(),
+            balance: payable_account_balance_1,
+            last_paid_timestamp: payment_timestamp_1,
+            pending_payment_rowid_opt: None,
+        };
+        let wallet_account_2 = make_wallet("creditor2");
+        let account_2 = PayableAccount {
+            wallet: wallet_account_2.clone(),
+            balance: payable_account_balance_2,
+            last_paid_timestamp: payment_timestamp_2,
+            pending_payment_rowid_opt: None,
+        };
+        let pending_payments_scan_interval = 200; //should be slightly less than 1/5 of the time until shutting the system
+        let payable_dao = PayableDaoMock::new()
+            .non_pending_payables_params(&non_pending_payables_params_arc)
+            .non_pending_payables_result(vec![account_1, account_2])
+            .mark_pending_payment_params(&mark_pending_payment_params_arc)
+            .mark_pending_payment_result(Ok(()))
+            .mark_pending_payment_result(Ok(()))
+            .transaction_confirmed_params(&transaction_confirmed_params_arc)
+            .transaction_confirmed_result(Ok(()));
+        let bootstrapper_config = bc_from_ac_plus_earning_wallet(
+            AccountantConfig {
+                payables_scan_interval: Duration::from_secs(1_000_000), //we don't care about this scan
+                receivables_scan_interval: Duration::from_secs(1_000_000), //we don't care about this scan
+                pending_payments_scan_interval: Duration::from_millis(
+                    pending_payments_scan_interval,
+                ),
+                when_pending_too_long_sec: (PAYMENT_CURVES.payment_suggested_after_sec + 1000)
+                    as u64,
+            },
+            make_wallet("some_wallet_address"),
+        );
+        let payment_1_backup_first_round = PaymentBackupRecord {
+            rowid: rowid_for_account_1,
+            timestamp: payment_timestamp_1,
+            hash: pending_tx_hash_1,
+            attempt: 1,
+            amount: payable_account_balance_1 as u64,
+            process_error: None,
+        };
+        let payment_2_backup_first_round = PaymentBackupRecord {
+            rowid: rowid_for_account_2,
+            timestamp: payment_timestamp_2,
+            hash: pending_tx_hash_2,
+            attempt: 1,
+            amount: payable_account_balance_2 as u64,
+            process_error: None,
+        };
+        let mut payment_1_backup_second_round = payment_1_backup_first_round.clone();
```

Comment:
Memory is hazy, but I may remember a Rust way to do this more concisely. Maybe involving two or three dots in a row?